Demo showcasing lifecycle methods

hopsoft opened the pull request:
Newer version of #25 using latest API

Demo: https://monosnap.com/file/JAUYNvhyEJFi24UnQ6x4mfS7mDeDDw

Lines of custom JavaScript: 45
Total lines of JavaScript: 66

pjforde1978 commented:
It's so great to see this working. Great progress.

I went down a rabbit hole / k-hole of classic Rails videos tonight. One of the things that really blew my mind is the degree to which DHH's 15 minute video delivers on its promise. In fact, now that there's migrations the video could have probably been 12 minutes. I bring this up because I saw [this](https://www.youtube.com/watch?v=f-qY37JIdg0) and it pissed me off. How does it seem okay to anyone that in 15 years of thousands of people working on this stuff, that it now takes 30 or 40 minutes to do far less than what she shows, given that she skipped a lot of steps. I don't mean to be harsh on her individually, but it's frankly embarrassing that anyone could look at all of the insane BS she has to know and do to get a less-functional CRUD app working. I'd say that it's a conspiracy if it wasn't just a really sad example of engineers fixing what wasn't broken. As you know, I love Sam Stephenson's Turbolinks video and he talks a lot about how the request/response model was his favourite feature of early Rails. I kind of love the idea that stimulus_reflex gets us to an even better place. It got me to thinking that we could provide a generator that creates the reflexes necessary to handle basic CRUD that is built on the built in rails scaffold generator (is that still a thing?).

Related but unrelated: thanks for tipping me off to velocity-animate. What a throwback to Scriptaculous! One laughably small nitpick: the yellow highlight color code is 100% definitely supposed to be #ffff99. (I used to do a lot of Effect.Highlight back in the day.)

Given what you warned me about over Signal, I don't have much feedback beyond "cool". I noticed that you restructured everything from todos to todo. I saw that you have a lot of data-controller="todo" in the view templates... I assume that's a temporary thing to support the data-reflex declarations? I definitely hope that you don't have to call data-controller in order to have a data-reflex in the final build. Baby steps! I'm patient.

Anyhow, this is rambly and pointless so I'll wrap by saying that I'm loving where this is going. Great work.

hopsoft commented:
I've updated the demo to a much cleaner implementation. I moved all the visual effects to a controller dedicated to this concern and removed the superfluous `data-controller` attributes as they aren't necessary. It's starting to feel much closer to the correct way to structure StimulusReflex apps. Thoughts?

andrewmcodes commented:
I think this looks great. Much cleaner.
